### PR TITLE
feat(api): API improvements

### PR DESCRIPTION
**Pull request by jonbjarnio**
- Updated the API layer to match the changes in the application system
- Added validation factory method to handle errors.
- Removed unecessary lookup calls that will throw errors either way if we skip them.
- Added overall improvements and optimisations where possible.

### CODE REVIEW

File: libs/shared/pipelines/src/lib/exception-factory.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+        return {
+          property: error.property,
+          constraints: error.constraints,
+        }
+      })
+      return new BadRequestException(errs)
+    },
+    transform: true,
+    // stopAtFirstError: true,
```
Comment (thorkellmani) [inline libs/shared/pipelines/src/lib/exception-factory.pipe.ts:25]: 👀

**Review by thorkellmani — COMMENTED** [review]
Some nitpicks

File: libs/shared/modules/src/case/case.service.ts [hunk 2]
```diff
@@ -146,32 +237,30 @@ export class CaseService implements ICaseService {
...
         involvedPartyId: activeCase.involvedParty.id,
-        categoryIds: activeCase.advertCategories.map((c) => c.id),
+        categoryIds: activeCase.categories.map((c) => c.id),
         statusId: advertStatus.id,
         subject: activeCase.advertTitle,
         publicationNumber: number,
-        publicationDate: new Date(),
-        signatureDate: new Date(advert.signatureDate),
+        publicationDate: now,
+        signatureDate: now, // TODO: Replace with signature
```
[the remark this answers is not in the document]
Comment (jonbjarnio) [reply libs/shared/modules/src/case/case.service.ts:250]: Will be resolved later

File: libs/shared/modules/src/case/case.service.ts [hunk 3]
```diff
@@ -431,11 +436,43 @@ export class CaseService implements ICaseService {
...
+        async (channel) =>
+          await this.createCaseChannel(
+            caseId,
+            {
+              email: channel.email,
+              phone: channel.phone,
+            },
+            channelsTransaction,
+          ),
+      )
```
[the remark this answers is not in the document]
Comment (jonbjarnio) [reply libs/shared/modules/src/case/case.service.ts:469]: Do mean bulk create, or do you mean creating array of promises?

File: libs/shared/modules/src/case/case.service.ts [hunk 4]
```diff
@@ -461,26 +498,47 @@ export class CaseService implements ICaseService {
     // TODO: ApplicationCommunicationChannels?
```
[the remark this answers is not in the document]
Comment (jonbjarnio) [reply libs/shared/modules/src/case/case.service.ts:498]: Will be resolved in future PR's

**Pull request merged**